open_humans/account_views.py

```diff
@@ -297,11 +288,15 @@ def dispatch(self, request, *args, **kwargs):
             extra_data = self.sociallogin.account.extra_data
         except AttributeError:
             return ret
-        email = extra_data['email']
-        if email_address_exists(email):
+        self.email = extra_data['email']
+        if email_address_exists(self.email):
             if not self.sociallogin.is_existing:
-                self.sociallogin.user = EmailAddress.objects.get(
-                    email=email).user
+                account_emailaddress = EmailAddress.objects.get(
```

Comment:
The docstring says this method is overridden to "just log in if email exists" – but it seems more is happening here, is that correct?

But... **I'm going to question why this class – and `MemberSignupView` – exist in the first place?**

Did you look into using a custom signup form class as described by `django-allauth`'s configuration documentation? https://django-allauth.readthedocs.io/en/latest/configuration.html

In general we should be using the customization methods a package is designed to support. And in particular, overriding a `dispatch` method is a very aggressive thing to be doing to a View.

  * `django-allauth` supports defining a `ACCOUNT_SIGNUP_FORM_CLASS` that can be used for customizations – it explicitly describes using it to process additional fields like a newsletter, and should define a `def signup(self, request, user)` method. Can that conceivably used for everything covered by both of these custom views?
  * Overriding a view is not necessary to override the template: just put the template in the right place, and it is overridden.
  * The login redirect URL (on success) should be defined by `LOGIN_REDIRECT_URL` (a general Django thing)

Reply:
Have plans to refactor this.

Reply:
@jhdulaney did this refactoring happen?